File: android/app/src/main/java/com/vegettable/app/model/ProductSummary.java
package com.vegettable.app.model;

import com.google.gson.annotations.SerializedName;
import java.util.List;

public class ProductSummary {
    @SerializedName("cropCode")
    private String cropCode;

    @SerializedName("cropName")
    private String cropName;

    @SerializedName("avgPrice")
    private double avgPrice;

    @SerializedName("prevAvgPrice")
    private double prevAvgPrice;

    @SerializedName("historicalAvgPrice")
    private double historicalAvgPrice;

    @SerializedName("volume")
    private double volume;

    @SerializedName("priceLevel")
    private String priceLevel;

    @SerializedName("trend")
    private String trend;

    @SerializedName("recentPrices")
    private List<DailyPrice> recentPrices;

    @SerializedName("category")
    private String category;

    @SerializedName("subCategory")
    private String subCategory;

    @SerializedName("aliases")
    private List<String> aliases;

    // Getters
    public String getCropCode() { return cropCode; }
    public String getCropName() { return cropName; }
    public double getAvgPrice() { return avgPrice; }
    public double getPrevAvgPrice() { return prevAvgPrice; }
    public double getHistoricalAvgPrice() { return historicalAvgPrice; }
    public double getVolume() { return volume; }
    public String getPriceLevel() { return priceLevel; }
    public String getTrend() { return trend; }
    public List<DailyPrice> getRecentPrices() { return recentPrices; }
    public String getCategory() { return category; }
    public String getSubCategory() { return subCategory; }
    public List<String> getAliases() { return aliases; }
}

File: android/app/src/main/java/com/vegettable/app/model/ProductSummary.kt
package com.vegettable.app.model

import com.google.gson.annotations.SerializedName

class ProductSummary {
    // Getters
    @JvmField
    @SerializedName("cropCode")
    val cropCode: String? = null

    @JvmField
    @SerializedName("cropName")
    val cropName: String? = null

    @JvmField
    @SerializedName("avgPrice")
    val avgPrice: Double = 0.0

    @SerializedName("prevAvgPrice")
    val prevAvgPrice: Double = 0.0

    @SerializedName("historicalAvgPrice")
    val historicalAvgPrice: Double = 0.0

    @SerializedName("volume")
    val volume: Double = 0.0

    @JvmField
    @SerializedName("priceLevel")
    val priceLevel: String? = null

    @JvmField
    @SerializedName("trend")
    val trend: String? = null

    @SerializedName("recentPrices")
    val recentPrices: MutableList<DailyPrice?>? = null

    @SerializedName("category")
    val category: String? = null

    @SerializedName("subCategory")
    val subCategory: String? = null

    @JvmField
    @SerializedName("aliases")
    val aliases: MutableList<String?>? = null
}
